app/controllers/volunteer_tasks_controller.rb

```diff
@@ -101,16 +101,30 @@ def edit
     @tasks_categories = %w[Events Projects Supervising Workshops Other]
     @certifications = Training.all
     @volunteers_in_task =
-      User.joins(:programs).where(
-        programs: {
-          program_type: Program::VOLUNTEER
-        },
-        id: @volunteer_task.volunteer_task_joins.pluck(:user_id)
+      User.where(
+        id:
+          VolunteerTaskJoin
+            .where(volunteer_task_id: @volunteer_task.id)
+            .where(user_type: "Volunteer")
+            .pluck(:user_id)
       )
     @staff_in_task =
-      User.where("users.role = ? OR users.role = ?", "staff", "admin").where(
-        id: @volunteer_task.volunteer_task_joins.pluck(:user_id)
+      User.where(
+        id:
+          VolunteerTaskJoin
+            .where(volunteer_task_id: @volunteer_task.id)
+            .where.not(user_type: "Volunteer")
+            .pluck(:user_id)
       )
+    @staff_available =
+      User
+        .where("users.role = ? OR users.role = ?", "staff", "admin")
```

Comment:
Would this work? `"users.role = staff OR users.role = admin"`, it would be better for SQL injection